frontend/readcollection.lua

```diff
@@ -0,0 +1,156 @@
+local DataStorage = require("datastorage")
+local LuaSettings = require("luasettings")
+local getFriendlySize = require("util").getFriendlySize
+local lfs = require("libs/libkoreader-lfs")
+local realpath = require("ffi/util").realpath
+local util = require("frontend/util")
+
+local collection_file = DataStorage:getSettingsDir() .. "/collection.lua"
+
+local ReadCollection = {}
+
+function ReadCollection:read(collections)
+    local collection_handle = LuaSettings:open(collection_file)
+    local coll = collection_handle:readSetting(collections) or {}
```

Comment:
That file was a bit hard to understand :)

I guess that `collections` (plural) here , `collection` (singular) below is a _collection name_, and you pass "favorites" as a string when you call it (your single default collection name for now, as I understand we'll be able to manage more later).
So, may be use `collection_name` everywhere as the variable holding that collection name string, to make it easier to understand.
And use `collections` (plural) when you're dealing with the whole datastructure. And `collection` (singular) when dealing with a sub-table of a specific `collection_name`.

And may be, to avoid passing "favorites" everywhere else, just pass nothing, and in here:
```lua
local DEFAULT_COLLECTION_NAME = "favorites"
[...]
function ReadCollection:something(collection_name)
  if not collection_name then collection_name = DEFAULT_COLLECTION_NAME end
```